src/datasources/config-api/config-api.service.ts

```diff
@@ -13,20 +16,37 @@ export class ConfigApi {
     private readonly dataSource: CacheFirstDataSource,
     @Inject(IConfigurationService)
     private readonly configurationService: IConfigurationService,
+    private readonly validationErrorFactory: ValidationErrorFactory,
   ) {
     this.baseUri =
       this.configurationService.getOrThrow<string>('safeConfig.baseUri');
   }
 
   async getChains(): Promise<Page<Chain>> {
     const key = 'chains'; // TODO key is not final
-    const url = this.baseUri + '/api/v1/chains';
-    return await this.dataSource.get(key, url);
+    const url = `${this.baseUri}/api/v1/chains`;
+    const page: Page<Chain> = await this.dataSource.get(key, url);
+
+    if (!page?.results.every((chain) => isValidChain(chain))) {
```

Comment:
`every` seems to test every element of the array. Do we want that? As in, is it important for the error collection?

We can also consider `some` if we want to exit "early".

Reply:
I think both `some` and `every` methods return immediately if the condition is met. From MDN docs:
```
The every method executes the provided callbackFn function once for each element present in the array until it finds the one where callbackFn returns a [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value. If such an element is found, the every method immediately returns false
```

Maybe `some` is more readable in this case? Wdyt?
```
if (page?.results.some((chain) => !isValidChain(chain)))
```

Reply:
Ah my bad. The way you wrote it good! No need to change it 👍 

I was just considering if we'd fail early which seems to be the case. Thanks for explaining!